src/plone/restapi/deserializer/__init__.py

```diff
@@ -4,6 +4,9 @@
 
 
 def json_body(request):
+    # TODO We should not read the complete request BODY in memory.
+    # Once we have fixed this, we can remove the temporary patches.py.
+    # See there for background information.
     try:
         data = json.loads(request.get("BODY") or "{}")
```

Comment:
Based on https://github.com/zopefoundation/Zope/blob/master/src/ZPublisher/HTTPRequest.py#L1061, I think we could do (untested):
```
body = request.get("BODYFILE")
data = {} if body is None else json.load(body)
```

This will not actually avoid reading the file into memory (https://pythonspeed.com/articles/json-memory-streaming/ makes it clear that json.load still does that) but would bypass the descriptor that enforces the VALUE_LIMIT.

Reply:
I thought `BODYFILE` would only be defined when an actual file is being uploaded, and not for example when you POST a login and password.  But that is not true.

It does not sound like a permanent solution, as this would still offer a way to potentially DOS the server.  But maybe very large uploads would still get stopped by one of the other limits.  And initial testing seems to work out.

Let me open a different PR, so we still have the current one in case there are problems.